packages/core/src/modules/proofs/ProofsApi.ts

```diff
@@ -555,29 +559,59 @@ export class ProofsApi<PPs extends ProofProtocol[]> implements ProofsApi<PPs> {
    */
   public async sendProblemReport(options: SendProofProblemReportOptions): Promise<ProofExchangeRecord> {
     const proofRecord = await this.getById(options.proofRecordId)
-    if (!proofRecord.connectionId) {
-      throw new AriesFrameworkError(`No connectionId found for proof record '${proofRecord.id}'.`)
-    }
 
     const protocol = this.getProtocol(proofRecord.protocolVersion)
-    const connectionRecord = await this.connectionService.getById(this.agentContext, proofRecord.connectionId)
 
-    // Assert
-    connectionRecord.assertReady()
+    const requestMessage = await protocol.findRequestMessage(this.agentContext, proofRecord.id)
 
     const { message: problemReport } = await protocol.createProblemReport(this.agentContext, {
       proofRecord,
       description: options.description,
     })
 
-    const outboundMessageContext = new OutboundMessageContext(problemReport, {
-      agentContext: this.agentContext,
-      connection: connectionRecord,
-      associatedRecord: proofRecord,
-    })
+    if (proofRecord.connectionId) {
+      const connectionRecord = await this.connectionService.getById(this.agentContext, proofRecord.connectionId)
 
-    await this.messageSender.sendMessage(outboundMessageContext)
-    return proofRecord
+      // Assert
+      connectionRecord.assertReady()
+
+      const outboundMessageContext = new OutboundMessageContext(problemReport, {
```

Comment:
@TimoGlastra is this the correct place to do this logic? If possbile I think it would be good to defer this to another method (creating the service decorator and adding it as `serviceParams`).

Reply:
The logical changes for `send Problem Report` function (handle connectionless request) repeats logic from `accept Request` method.